pkg/operator/observe_config.go

```diff
@@ -84,22 +113,66 @@ func (c ConfigObserver) sync() error {
 	}
 	if len(etcdURLs) > 0 {
 		unstructured.SetNestedStringSlice(observedConfig, etcdURLs, "storageConfig", "urls")
+	} else {
+		glog.Warningf("no etcd endpoints found")
 	}
+	return observedConfig, nil
+}
 
-	// don't worry about errors
-	currentConfig := map[string]interface{}{}
-	json.NewDecoder(bytes.NewBuffer(operatorConfig.Spec.ObservedConfig.Raw)).Decode(&currentConfig)
-	if reflect.DeepEqual(currentConfig, observedConfig) {
-		return nil
+func (c *ConfigObserver) observeClusterConfig(observedConfig map[string]interface{}) (map[string]interface{}, error) {
+
+	// observe CIDRs from cluster-config-v1 in order to populate list of restrictedCIDRs
+
+	clusterConfig, err := c.kubeClient.CoreV1().ConfigMaps("kube-system").Get("cluster-config-v1", metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return observedConfig, fmt.Errorf("error retieving cluster config: %s", err)
```

Comment:
never wrap errors.  Wrapping errors works really poorly because it destroys the ability to determine the type of error higher in the stack.